crates/ruff_linter/resources/test/fixtures/pylint/self_cls_assignment.py

```diff
@@ -0,0 +1,18 @@
+class Fruit:
+    @classmethod
+    def list_fruits(cls) -> None:
+        cls = "apple"  # PLW0642
+        cls: Fruit = "apple"  # PLW0642
+        cls += "orange"  # PLW0642
+        cls, blah = "apple", "orange"  # PLW0642
+        blah, (cls, blah2) = "apple", ("orange", "banana")  # PLW0642
```

Comment:
We should also add test cases when the assignment target is a list. Refer to my comment below.